core/internal/common.go

```diff
@@ -0,0 +1,52 @@
+// Copyright 2017 Google Inc. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package internal
+
+import (
+	"fmt"
+
+	"github.com/golang/protobuf/ptypes"
+	"github.com/golang/protobuf/ptypes/any"
+
+	tpb "github.com/google/keytransparency/core/proto/keytransparency_v1_types"
+	"github.com/google/trillian"
+)
+
+// MetadataFromMapRoot converts the google.protobuf.Any metadata field from the
+// Trillian SignedMapRoot into a tpb.MapperMetadata, or returns an error if the
+// field is absent or of the wrong type.
+func MetadataFromMapRoot(r *trillian.SignedMapRoot) (*tpb.MapperMetadata, error) {
+	if r.GetMetadata() == nil {
+		return &tpb.MapperMetadata{}, nil
+	}
+	var metadataProto ptypes.DynamicAny
+	if err := ptypes.UnmarshalAny(r.GetMetadata(), &metadataProto); err != nil {
+		return nil, fmt.Errorf("Failed to unmarshal MapRoot.Metadata: %v", err)
+	}
+	return metadataProto.Message.(*tpb.MapperMetadata), nil
+}
+
+// MetadataAsAny marshals the supplied MapperMetadata proto into an 'any', which
+// can be supplied as metadata to the Trillian map when setting leaves.
+func MetadataAsAny(meta *tpb.MapperMetadata) (*any.Any, error) {
+	if meta == nil {
+		return &any.Any{}, nil
```

Comment:
I'm a little unsure about the correctness of `any.Any{}`.  Should we be marshaling an empty `MapperMetadata` here?

Reply:
The thinking was that we should always provide an `any.Any` when setting map leaves, and in the case where there's no metadata to store, that the 'nil' value for any.Any is used.
But I think you're right, it'd be better to set meta = &tpb.MapperMetadata{} in this branch, then continue.

Reply:
Done.